src/git/zcl_abapgit_git_transport.clas.abap

```diff
@@ -7,17 +7,16 @@ CLASS zcl_abapgit_git_transport DEFINITION
 
 * remote to local
     CLASS-METHODS upload_pack
-      IMPORTING
-        !iv_url         TYPE string
-        !iv_branch_name TYPE string
-        !iv_deepen      TYPE abap_bool DEFAULT abap_true
-        !it_branches    TYPE zif_abapgit_definitions=>ty_git_branch_list_tt OPTIONAL
-      EXPORTING
-        !et_objects     TYPE zif_abapgit_definitions=>ty_objects_tt
-        !ev_branch      TYPE zif_abapgit_definitions=>ty_sha1
-        !eo_branch_list TYPE REF TO zcl_abapgit_git_branch_list
-      RAISING
-        zcx_abapgit_exception .
+      IMPORTING iv_url          TYPE string
+                iv_branch_name  TYPE string
+                iv_commit_hash  TYPE zif_abapgit_definitions=>ty_sha1 OPTIONAL
+                iv_deepen_level TYPE numc2 DEFAULT '0'
```

Comment:
what is the background for changing the logic regarding deepen? This PR concerns getting remote objects by commit hash, so I do not expect the deepen logic to change

Reply:
I call the git api with deepen_level = 99 to select as much commit hashes as possible and after that load them into the popup to be displayed to be able to select one one want to revert to. In my first concept I tried to get them by its hierarchie but unfortunately sometimes commits seem to have no parent I would know at one step in the hierarchie so I would have to select "blindly" to step down the history.